### ISSUE TITLE
Download Problem

### ISSUE THREAD

**Issue by raphuss** [moisesmcardona/GoogleDriveUploadTool#8]
I like this project really, thanks for this program. Its uploading with no problem. Just a little thing. It may be good if it has an estimated time left feauture like drive. I mean a total time to finish uploading, not time to finish folder by folder. 

I wanted to say it seems there are some problems on downloading. I am attaching images telling this problem. If you can find a solution, please make me know.

<img width="662" alt="1" src="https://user-images.githubusercontent.com/36515485/36273533-cf7f41a6-1295-11e8-9608-3bbd69714d54.png">
<img width="206" alt="2" src="https://user-images.githubusercontent.com/36515485/36273534-cfa26582-1295-11e8-96a7-c8547839b207.png">

**Comment by moisespr123** [comment]
Hi,

Thanks for letting me know of this issue. I just took a look into this and applied a fix in the code. Will release an executable later.

**Comment by raphuss** [comment]
Thanks, if you have time, fixing estimated time issue also would be great. Will check next exe and give feedbacks if needed.

**Comment by raphuss** [comment]
Hi again, thanks for changing. I tested new version and downloading without any problem now. But cant we add a "folder downloading" option into this program?

**Comment by moisespr123** [comment]
Yes, that will be added. I've been making some commits to simplify the code to prepare for the folder download code. Expect that too!

**Update**: Feature has been added and tested in code.

**Comment by raphuss** [comment]
Thanks, will be waiting for it.

Update: Testing now.

**Comment by raphuss** [comment]
I think there is an error on forms.vb about fileid parameter. Error i took is below:

```
The service drive has thrown an exception: Google.GoogleApiException: Parameter "fileId" is missing
   path: Google.Apis.Requests.ClientServiceRequest`1.AddParameters(RequestBuilder requestBuilder, ParameterCollection inputParameters)
   path: Google.Apis.Requests.ClientServiceRequest`1.CreateBuilder()
   path: Google.Apis.Requests.ClientServiceRequest`1.GenerateRequestUri()
   path: Google.Apis.Drive.v3.FilesResource.GetRequest.DownloadAsync(Stream stream)
   path: Google_Drive_Uploader.Form1.VB$StateMachine_39_DownloadFile.MoveNext() C:\Users\*username*\Desktop\GoogleDriveUploadTool-master\Google Drive Uploader1\Form1.vb içinde: row 398
--- Özel durumun oluşturulduğu önceki konumdan başlayan yığın izlemesinin sonu ---
   path: System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   path: System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   path: System.Runtime.CompilerServices.TaskAwaiter.GetResult()
   path: Google_Drive_Uploader.Form1.VB$StateMachine_38_Button5_Click.MoveNext() C:\Users\*username*\Desktop\GoogleDriveUploadTool-master\Google Drive Uploader1\Form1.vb içinde: row 387
--- Özel durumun oluşturulduğu önceki konumdan başlayan yığın izlemesinin sonu ---
   path: System.Runtime.CompilerServices.AsyncMethodBuilderCore.<>c.<ThrowAsync>b__6_0(Object state)

************** Installed Compilations **************
mscorlib
    Bütünleştirilmiş Kod Sürümü: 4.0.0.0
    Win32 Sürümü: 4.7.2563.0 built by: NET471REL1
    CodeBase: file:///C:/Windows/Microsoft.NET/Framework/v4.0.30319/mscorlib.dll
----------------------------------------
Google Drive Uploader1
    Bütünleştirilmiş Kod Sürümü: 1.7.0.0
    Win32 Sürümü: 1.7.0.0
    CodeBase: file:///C:/Users/*username*/Desktop/GoogleDriveUploadTool-master/Google%20Drive%20Uploader1/bin/Debug/Google%20Drive%20Uploader1.exe
----------------------------------------
Microsoft.VisualBasic
    Bütünleştirilmiş Kod Sürümü: 10.0.0.0
    Win32 Sürümü: 14.7.2558.0 built by: NET471REL1
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/Microsoft.VisualBasic/v4.0_10.0.0.0__b03f5f7f11d50a3a/Microsoft.VisualBasic.dll
----------------------------------------
System
    Bütünleştirilmiş Kod Sürümü: 4.0.0.0
    Win32 Sürümü: 4.7.2558.0 built by: NET471REL1
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System/v4.0_4.0.0.0__b77a5c561934e089/System.dll
----------------------------------------
System.Core
    Bütünleştirilmiş Kod Sürümü: 4.0.0.0
    Win32 Sürümü: 4.7.2563.0 built by: NET471REL1
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Core/v4.0_4.0.0.0__b77a5c561934e089/System.Core.dll
----------------------------------------
System.Windows.Forms
    Bütünleştirilmiş Kod Sürümü: 4.0.0.0
    Win32 Sürümü: 4.7.2558.0 built by: NET471REL1
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Windows.Forms/v4.0_4.0.0.0__b77a5c561934e089/System.Windows.Forms.dll
----------------------------------------
System.Drawing
    Bütünleştirilmiş Kod Sürümü: 4.0.0.0
    Win32 Sürümü: 4.7.2558.0 built by: NET471REL1
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Drawing/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Drawing.dll
----------------------------------------
System.Configuration
    Bütünleştirilmiş Kod Sürümü: 4.0.0.0
    Win32 Sürümü: 4.7.2558.0 built by: NET471REL1
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Configuration/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Configuration.dll
----------------------------------------
System.Xml
    Bütünleştirilmiş Kod Sürümü: 4.0.0.0
    Win32 Sürümü: 4.7.2612.0 built by: NET471REL1LAST_B
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Xml/v4.0_4.0.0.0__b77a5c561934e089/System.Xml.dll
----------------------------------------
System.Runtime.Remoting
    Bütünleştirilmiş Kod Sürümü: 4.0.0.0
    Win32 Sürümü: 4.7.2558.0 built by: NET471REL1
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Runtime.Remoting/v4.0_4.0.0.0__b77a5c561934e089/System.Runtime.Remoting.dll
----------------------------------------
Google.Apis.Drive.v3
    Bütünleştirilmiş Kod Sürümü: 1.32.2.1135
    Win32 Sürümü: 1.32.2.1135
    CodeBase: file:///C:/Users/*username*/Desktop/GoogleDriveUploadTool-master/Google%20Drive%20Uploader1/bin/Debug/Google.Apis.Drive.v3.DLL
----------------------------------------
Google.Apis
    Bütünleştirilmiş Kod Sürümü: 1.32.2.0
    Win32 Sürümü: 1.32.2.0
    CodeBase: file:///C:/Users/*username*/Desktop/GoogleDriveUploadTool-master/Google%20Drive%20Uploader1/bin/Debug/Google.Apis.DLL
----------------------------------------
Google.Apis.Core
    Bütünleştirilmiş Kod Sürümü: 1.32.2.0
    Win32 Sürümü: 1.32.2.0
    CodeBase: file:///C:/Users/*username*/Desktop/GoogleDriveUploadTool-master/Google%20Drive%20Uploader1/bin/Debug/Google.Apis.Core.DLL
----------------------------------------
Google.Apis.Auth
    Bütünleştirilmiş Kod Sürümü: 1.32.2.0
    Win32 Sürümü: 1.32.2.0
    CodeBase: file:///C:/Users/*username*/Desktop/GoogleDriveUploadTool-master/Google%20Drive%20Uploader1/bin/Debug/Google.Apis.Auth.DLL
----------------------------------------
System.Net.Http
    Bütünleştirilmiş Kod Sürümü: 4.0.0.0
    Win32 Sürümü: 4.7.2558.0 built by: NET471REL1
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Net.Http/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Net.Http.dll
----------------------------------------
Microsoft.GeneratedCode
    Bütünleştirilmiş Kod Sürümü: 1.0.0.0
    Win32 Sürümü: 4.7.2612.0 built by: NET471REL1LAST_B
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Xml/v4.0_4.0.0.0__b77a5c561934e089/System.Xml.dll
----------------------------------------
Newtonsoft.Json
    Bütünleştirilmiş Kod Sürümü: 10.0.0.0
    Win32 Sürümü: 10.0.3.21018
    CodeBase: file:///C:/Users/*username*/Desktop/GoogleDriveUploadTool-master/Google%20Drive%20Uploader1/bin/Debug/Newtonsoft.Json.DLL
----------------------------------------
System.Numerics
    Bütünleştirilmiş Kod Sürümü: 4.0.0.0
    Win32 Sürümü: 4.7.2558.0 built by: NET471REL1
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Numerics/v4.0_4.0.0.0__b77a5c561934e089/System.Numerics.dll
----------------------------------------
System.Runtime.Serialization
    Bütünleştirilmiş Kod Sürümü: 4.0.0.0
    Win32 Sürümü: 4.7.2623.0 built by: NET471REL1LAST_C
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Runtime.Serialization/v4.0_4.0.0.0__b77a5c561934e089/System.Runtime.Serialization.dll
----------------------------------------
System.Data
    Bütünleştirilmiş Kod Sürümü: 4.0.0.0
    Win32 Sürümü: 4.7.2623.0 built by: NET471REL1LAST_C
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_32/System.Data/v4.0_4.0.0.0__b77a5c561934e089/System.Data.dll
----------------------------------------
mscorlib.resources
    Bütünleştirilmiş Kod Sürümü: 4.0.0.0
    Win32 Sürümü: 4.7.2558.0 built by: NET471REL1
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/mscorlib.resources/v4.0_4.0.0.0_tr_b77a5c561934e089/mscorlib.resources.dll
----------------------------------------
System.Windows.Forms.resources
    Bütünleştirilmiş Kod Sürümü: 4.0.0.0
    Win32 Sürümü: 4.7.2558.0 built by: NET471REL1
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Windows.Forms.resources/v4.0_4.0.0.0_tr_b77a5c561934e089/System.Windows.Forms.resources.dll
----------------------------------------

************** JIT Debugging**************
To activate JIT debugging jitDebugging value should be defined on system.windows.forms part of .config file of this computer(machine.config) or this software.  Also software needed to be compiled when debugging is active.

Example:

<configuration>
    <system.windows.forms jitDebugging="true" />
</configuration>
```

**Comment by moisespr123** [comment]
It's working fine for me! Please note this will not download Google Docs files. Only actual files and create the same folder structure as in Google Drive.

I've released v1.7 with the download folder functionality added!

Further issues can be opened in a new issue ticket.

As for the estimate time, I'll research that deeper and see if I can come to a better estimate timer.

Enjoy!
